src/libs/ReportUtils.ts

```diff
@@ -4618,6 +4625,61 @@ function shouldDisableThread(reportAction: OnyxEntry<ReportAction>, reportID: st
     );
 }
 
+function getAllAncestorReportActions(
+    report: Report | null | undefined,
+    shouldHideThreadDividerLine: boolean,
+    reports: OnyxCollection<Report> = {},
+    reportActions: OnyxCollection<ReportActions> = {},
+): Ancestor[] {
+    if (!report) {
+        return [];
+    }
+    const convertReports: OnyxCollection<Report> = {};
+    const convertReportActions: OnyxCollection<ReportActions> = {};
+    Object.values(reports ?? {}).forEach((itemReport) => {
+        if (!itemReport) {
+            return;
+        }
+        convertReports[itemReport.reportID] = itemReport;
+    });
+    Object.keys(reportActions ?? {}).forEach((actionKey) => {
+        if (!actionKey) {
+            return;
+        }
+        const reportID = CollectionUtils.extractCollectionItemID(actionKey as `reportActions_${string}`);
+        convertReportActions[reportID] = reportActions?.[actionKey] ?? null;
+    });
```

Comment:
We are taking an extra step to iterate over all the reports and report actions. Can we try to do this in a single iteration? I think it is possible.

Reply:
Actually, we don't need the convert iteration there. We can replace the converted result `convertedReports` and `convertedReportActions` inside the while loop with this.

```
        const parentReport = reports?.[`${ONYXKEYS.COLLECTION.REPORT}${parentReportID}`];
        const parentReportAction = reportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`]?.[parentReportActionID ?? ''] ?? null;

        ...

        const isParentReportActionUnread = ReportActionsUtils.isCurrentActionUnread(parentReport, parentReportAction, reportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`] ?? {});
```

@dukenv0307 Let me know if that works.